### PR TITLE
feat: Add invite user API.

### PR DESCRIPTION
**Pull request by defectiveAi**

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.Tests/Model/PhoneNumberTests.cs [hunk 1]
```diff
@@ -33,6 +33,8 @@ public sealed class PhoneNumberTests
         [InlineData("+44 7222 555 555", true)]
         [InlineData("+49 (173) 1799 806-44", true)]
         [InlineData("010101 letters 02", false)]
+        [InlineData("000000000000000000000000000000", true)]
+        [InlineData("0000000000000000000000000000001", false)]
         public void Ctor_Email_ValidatesAddress(string value, bool isValid)
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.Tests/Model/PhoneNumberTests.cs:38]: This seems wrong, either the name of the function is wrong, or the data is wrong. :)

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserInvitation.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        Email = email;
+        FirstName = firstName;
+        LastName = lastName;
+        PhoneNumber = phoneNumber;
+        RequiredAuthentication = requiredAuthentication;
+        AssignedActor = assignedActor;
+        AssignedRoles = assignedRoles;
+
+        ValidateActor();
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserInvitation.cs:42]: All exceptions in constructor are equal, as the object will never be constructed. So assigning at the top or at the bottom does not matter from outside the object.

Placing validation after assignment can help with reuse, as all the properties are available: easier to call Validate from future methods + write validation logic between properties.

I have seen both usages though and it is minor, so if you want the change, I can do it.

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserIdentity.cs [hunk 3]
```diff
@@ -13,13 +13,78 @@
...
+        Id = new ExternalUserId(Guid.Empty);
+        Email = email;
+        Status = UserStatus.Active;
+        FirstName = firstName;
+        LastName = lastName;
+        PhoneNumber = phoneNumber;
+        CreatedDate = DateTimeOffset.UtcNow;
+        Authentication = authentication;
+
+        ValidateName();
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserIdentity.cs:63]: See other comment.

File: source/Energinet.DataHub.MarketParticipant.Tests/Model/PhoneNumberTests.cs (hunk 1 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.MarketParticipant.Tests/Model/PhoneNumberTests.cs]: :O Fixed

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserInvitation.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/UserInvitation.cs:42]: Actually your argument is solid, So i see no reason to change this.

**Review by FirestarJes — DISMISSED** [review]

File: source/Energinet.DataHub.MarketParticipant.Application/Handlers/User/InviteUserHandler.cs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        _actorQueryRepository = actorQueryRepository;
+        _userRoleRepository = userRoleRepository;
+    }
+
+    public async Task<Unit> Handle(InviteUserCommand request, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        var assignedActor = await GetActorAsync(request.Invitation.AssignedActor).ConfigureAwait(false);
+        var assignedRoles = await GetUserRolesAsync(request).ConfigureAwait(false);
```
Comment (xnetsrak) [inline source/Energinet.DataHub.MarketParticipant.Application/Handlers/User/InviteUserHandler.cs:58]: Do you need to send full request command to method ?

**Review by xnetsrak — DISMISSED** [review]
one comment

File: source/Energinet.DataHub.MarketParticipant.Application/Handlers/User/InviteUserHandler.cs (hunk 4 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.MarketParticipant.Application/Handlers/User/InviteUserHandler.cs]: Fixed.

**Review by xnetsrak — APPROVED** [review]

**Review by xnetsrak — APPROVED** [review]

**Pull request merged**